packages/dds/test-dds-utils/src/ddsSnapshotHarness.ts

```diff
@@ -0,0 +1,88 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "node:assert";
+import { existsSync, readFileSync, writeFileSync } from "node:fs";
+import path from "node:path";
+
+// Simple filter to avoid tests with a name that would accidentally be parsed as directory traversal or other confusing things.
+const nameCheck = new RegExp(/^[^"/\\]+$/);
+
+let currentTestName: string | undefined;
+let currentTestFile: string | undefined;
+
+/**
+ * @internal
+ */
+export interface ISnapshotSuite {
+	/**
+	 * A utility function for setting up the current snapshot directory for tests.
+	 */
+	useSnapshotDirectory: () => void;
+
+	/**
+	 * Takes a snapshot of data and writes it to a file.
+	 * @param data - The data to take a snapshot of.
+	 * @param writeCompatible - A flag indicating whether the snapshot should be checked for consistency
+	 * with old-format data. Defaults to true.
+	 */
+	takeSnapshot: (data: string, writeCompatible?: boolean) => string;
+}
+
+function takeSnapshot(data: string, writeCompatible: boolean = true): string {
```

Comment:
since the snapshot function is scoped to the return of `createSnapshotSuite` semantically, you should move this as well as relevant locals (`currentTestName`, `currentTestFile`) to inside the body of that function.